### ISSUE TITLE
Typescript: No "exports" main defined

### ISSUE THREAD

**Issue by frankforpresident** [achingbrain/mortice#7]
Trying to convert my own project to typescript and I'm getting an error from this package, Using version ^3.0.0

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in C:\Users\Project\node_modules\mortice\package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:453:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:669:7)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\Frank\Documents\Bitbucket\gitmanagement\dist\services\cloneService.js:10:48)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

**Comment by frankforpresident** [comment]
Workaround, I've downgraded back to v2.0.1 and added an empty declaration file

**Comment by djm181** [comment]
This isn't really a bug in mortice - now that it is a pure ES module you have a couple of options of how to import it. And your typescript project is probably still `require`ing mortice.

This gist has good info on solutions: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

**Comment by frankforpresident** [comment]
Not using `require` anymore for my typescript projects

I guess it is correct?
`import mortice, { Mortice } from 'mortice';`

I only get this error on run time. Typescript doesn't complain at all

Mortice is the only package that isn't working for me. Other mainstream packages are working as expected.

**Comment by djm181** [comment]
Do you know if any of your other packages are exported as ES6 modules? Moreover, what does your tsconfig.json look like for your project? If you have `module: 'commonjs'` then Typescript is going to be emitting `require`s in place of your imports - you can verify this by checking your emitted JS files.

**Comment by frankforpresident** [comment]
I do have commonjs enabled

```
{
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "importHelpers": true,
    "moduleResolution": "node", 
    "esModuleInterop": true,
    "sourceMap": true, 
    "outDir": "./dist", 
    "strict": true, 
    "forceConsistentCasingInFileNames": true,
    "allowSyntheticDefaultImports": true
  },
  "include": ["./src"],
  "exclude": ["node_modules", "dist"]
}
```

**Comment by djm181** [comment]
Well there you have it. That is the issue. That gist I linked goes into detail as to how to upgrade your project to use ESM itself. Good luck!

**Comment by frankforpresident** [comment]
I can have a look at it. It will be cleaner indeed but why is `require` not supported anymore? It is still a valid way of importing right?

**Comment by djm181** [comment]
No, that's the point - it is invalid to "require" an ES module. It must be imported.

**Comment by achingbrain** [comment]
Closing as the solution has been shown in https://github.com/achingbrain/mortice/issues/7#issuecomment-1105732910 - you need to use `import` not `require`.
